parachain/modules/ismp/bnb-pos/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#[warn(unused_imports)]
+#[warn(unused_variables)]
+extern crate alloc;
+
+use core::marker::PhantomData;
+
+use alloc::{boxed::Box, collections::BTreeMap, string::ToString, vec, vec::Vec};
+use bnb_pos_verifier::{
+    primitives::{compute_epoch, BnbClientUpdate, ConsensusState, EPOCH_LENGTH},
+    verify_bnb_header, VerificationResult,
+};
+use codec::{Decode, Encode};
+use ismp::{
+    consensus::{ConsensusClient, ConsensusStateId, StateCommitment, StateMachineClient},
+    error::Error,
+    host::{IsmpHost, StateMachine},
+    messaging::StateCommitmentHeight,
+};
+use ismp_sync_committee::EvmStateMachine;
+
+pub const BNB_CONSENSUS_ID: ConsensusStateId = *b"BNBP";
+
+pub struct BnbClient<H: IsmpHost>(PhantomData<H>);
+
+impl<H: IsmpHost> Default for BnbClient<H> {
+    fn default() -> Self {
+        Self(PhantomData)
+    }
+}
+
+impl<H: IsmpHost> Clone for BnbClient<H> {
+    fn clone(&self) -> Self {
+        Self(PhantomData)
+    }
+}
+
+impl<H: IsmpHost + Send + Sync + Default + 'static> ConsensusClient for BnbClient<H> {
+    fn verify_consensus(
+        &self,
+        _host: &dyn IsmpHost,
+        _consensus_state_id: ismp::consensus::ConsensusStateId,
+        trusted_consensus_state: Vec<u8>,
+        proof: Vec<u8>,
+    ) -> Result<(Vec<u8>, ismp::consensus::VerifiedCommitments), ismp::error::Error> {
+        let bnb_client_update = BnbClientUpdate::decode(&mut &proof[..]).map_err(|_| {
+            Error::ImplementationSpecific("Cannot decode bnb client update".to_string())
+        })?;
+
+        let mut consensus_state = ConsensusState::decode(&mut &trusted_consensus_state[..])
+            .map_err(|_| {
+                Error::ImplementationSpecific("Cannot decode trusted consensus state".to_string())
+            })?;
+
+        let current_epoch_block_number =
```

Comment:
I don't think this rotation block logic is correct.
You don't need to compute the epoch, 
You are meant to compare the next validator rotation block with the attested header block number, if the latter is greater you rotate the validator set then move on to verify the update